ambry-store/src/main/java/com/github/ambry/store/StatsBasedCompactionPolicy.java

```diff
@@ -64,7 +64,7 @@ public CompactionDetails getCompactionDetails(long totalCapacity, long usedCapac
                 blobStoreStats.getMaxBlobSize());
         if (bestCandidateToCompact != null) {
           details = new CompactionDetails(validDataSizeByLogSegment.getFirst(),
-              bestCandidateToCompact.getSegmentsToCompact());
+              bestCandidateToCompact.getSegmentsToCompact(), bestCandidateToCompact);
```

Comment:
The 2nd parameter seems to be redundant if we already pass `bestCandidateToCompact` in.

Reply:
In some tests, the 2nd parameters is not related to 3rd parameter, so we can't remove the 2nd one.